### PR TITLE
Change ListRepo and InspectRepo to include auth information, among other changes

### PR DESCRIPTION
**Pull request by msteffen**
1. ListRepo and InspectRepo, when called via the PFS GRPC API, automatically include auth info. This is in line with how both pachctl and especially the Pachyderm dashboard are currently designed (fixes https://github.com/pachyderm/pachyderm/issues/2276)
2. `pachctl list-repo` and `pachctl inspect-repo` now pretty-print auth information returned from the two calls, if auth is enabled in the cluster
3. Strip GRPC error text related to auth from PFS errors (fixes https://github.com/pachyderm/pachyderm/issues/2281)
4. CreateRepo now calls SetACL instead of SetScope (and SetACL now permits users to set an ACL that exclusively appoints themselves as the owner of repos that don't exist), which I believe resolves the consistency issues between the two calls (one of the items in https://github.com/pachyderm/pachyderm/issues/2153)

### CODE REVIEW

File: src/server/auth/server/api_server.go [hunk 1]
```diff
@@ -203,7 +204,8 @@ func (a *apiServer) getEnterpriseTokenState() (enterpriseclient.State, error) {
 	resp, err := pachClient.Enterprise.GetState(context.Background(),
 		&enterpriseclient.GetStateRequest{})
 	if err != nil {
-		return 0, fmt.Errorf("could not get Enterprise status: %s", err.Error())
+		return 0, fmt.Errorf("could not get Enterprise status: %s",
```
Comment (derekchiang) [inline src/server/auth/server/api_server.go]: Using `%s` with `.Error()` is an anti-pattern.  You can use `%v` with the `error` type directly.

File: src/server/pfs/server/driver.go [hunk 2]
```diff
@@ -192,7 +192,8 @@ func (d *driver) checkIsAuthorized(ctx context.Context, r *pfs.Repo, s auth.Scop
 	if err == nil && !resp.Authorized {
 		return &auth.NotAuthorizedError{Repo: r.Name, Required: s}
 	} else if err != nil && !auth.IsNotActivatedError(err) {
-		return fmt.Errorf("error during authorization check for operation on %s: %s", r.Name, err.Error())
+		return fmt.Errorf("error during authorization check for operation on \"%s\": %s",
```
Comment (derekchiang) [inline src/server/pfs/server/driver.go]: See comment above.

File: src/server/pfs/server/driver.go [hunk 3]
```diff
@@ -234,17 +235,24 @@ func (d *driver) createRepo(ctx context.Context, repo *pfs.Repo, provenance []*p
 			&auth.WhoAmIRequest{})
 		if err != nil {
 			if !auth.IsNotActivatedError(err) {
-				return fmt.Errorf("authorization error while creating repo \"%s\": %s", repo.Name, err.Error())
+				return fmt.Errorf("authorization error while creating repo \"%s\": %s",
+					repo.Name, grpcutil.StripGRPCCode(err).Error())
```
Comment (derekchiang) [inline src/server/pfs/server/driver.go]: Same here.

File: src/server/pfs/server/driver.go [hunk 4]
```diff
@@ -234,17 +235,24 @@ func (d *driver) createRepo(ctx context.Context, repo *pfs.Repo, provenance []*p
...
+				NewACL: &auth.ACL{
+					Entries: map[string]auth.Scope{
+						whoAmI.Username: auth.Scope_OWNER,
+					},
+				},
 			})
 			if err != nil {
-				return fmt.Errorf("could not create ACL for new repo \"%s\": %s", repo.Name, err.Error())
+				return fmt.Errorf("could not create ACL for new repo \"%s\": %s",
+					repo.Name, grpcutil.StripGRPCCode(err).Error())
```
Comment (derekchiang) [inline src/server/pfs/server/driver.go]: Same.

File: src/server/pfs/server/driver.go [hunk 5]
```diff
@@ -381,12 +389,14 @@ func (d *driver) inspectRepo(ctx context.Context, repo *pfs.Repo, includeAuth bo
 		resp, err := d.pachClient.AuthAPIClient.GetScope(auth.In2Out(ctx),
 			&auth.GetScopeRequest{Repos: []string{repo.Name}})
 		if err != nil && !auth.IsNotActivatedError(err) {
-			return nil, fmt.Errorf("error getting scope for %s: %s", repo.Name, err.Error())
-		}
-		if len(resp.Scopes) != 1 {
-			return nil, fmt.Errorf("unexpected result from GetScope(): %#v", resp)
+			return nil, fmt.Errorf("error getting scope for \"%s\": %s", repo.Name,
```
Comment (derekchiang) [inline src/server/pfs/server/driver.go]: Same.

File: src/server/pfs/server/driver.go [hunk 6]
```diff
@@ -436,7 +446,8 @@ nextRepo:
 					Repos: []string{repoName},
 				})
 			if err != nil && !auth.IsNotActivatedError(err) {
-				return nil, fmt.Errorf("error getting scopes: %s", err.Error())
+				return nil, fmt.Errorf("error getting scopes: %s",
```
Comment (derekchiang) [inline src/server/pfs/server/driver.go]: Same.

File: src/server/pfs/server/api_server.go [hunk 7]
```diff
@@ -69,14 +69,14 @@ func (a *apiServer) InspectRepo(ctx context.Context, request *pfs.InspectRepoReq
...
-	return a.driver.inspectRepo(ctx, request.Repo, request.IncludeAuth)
+	return a.driver.inspectRepo(ctx, request.Repo, true)
 }
 
 func (a *apiServer) ListRepo(ctx context.Context, request *pfs.ListRepoRequest) (response *pfs.ListRepoResponse, retErr error) {
 	func() { a.Log(request, nil, nil, 0) }()
 	defer func(start time.Time) { a.Log(request, response, retErr, time.Since(start)) }(time.Now())
 
-	repoInfos, err := a.driver.listRepo(ctx, request.Provenance, request.IncludeAuth)
+	repoInfos, err := a.driver.listRepo(ctx, request.Provenance, true)
```
Comment (derekchiang) [inline src/server/pfs/server/api_server.go]: Should we just update `driver.listRepo` if `includeAuth` is never gonna be `false`?

File: src/server/auth/server/api_server.go (hunk 1 above)
Comment (msteffen) [reply src/server/auth/server/api_server.go]: Fixed

File: src/server/pfs/server/driver.go (hunk 2 above)
Comment (msteffen) [reply src/server/pfs/server/driver.go]: Fixed

File: src/server/pfs/server/driver.go (hunk 3 above)
Comment (msteffen) [reply src/server/pfs/server/driver.go]: Fixed

File: src/server/pfs/server/driver.go (hunk 4 above)
Comment (msteffen) [reply src/server/pfs/server/driver.go]: Fixed

File: src/server/pfs/server/driver.go (hunk 5 above)
Comment (msteffen) [reply src/server/pfs/server/driver.go]: Fixed

File: src/server/pfs/server/driver.go (hunk 6 above)
Comment (msteffen) [reply src/server/pfs/server/driver.go]: Fixed

File: src/server/pfs/server/api_server.go (hunk 7 above)
Comment (msteffen) [reply src/server/pfs/server/api_server.go]: There are a few places where it's still false, notably createRepo (https://github.com/pachyderm/pachyderm/blob/9aa9641f1d4279ed6664e54874463e4f37d5b739/src/server/pfs/server/driver.go#L310) and deleteAll (https://github.com/pachyderm/pachyderm/blob/9aa9641f1d4279ed6664e54874463e4f37d5b739/src/server/pfs/server/driver.go#L1902)

**Pull request merged**